src/graph/matrix/delta_matrix.rs

```diff
@@ -0,0 +1,510 @@
+                    &m.matrix,
+                    &n.delta_minus,
+                    GxB_ANY_PAIR_BOOL,
+                    null_mut(),
+                );
+                if mask.nvals() > 0 {
+                    (Some(mask), GrB_DESC_RSC)
+                } else {
+                    (None, null_mut())
+                }
+            } else {
+                (None, null_mut())
+            };
+
+            let accum = if n.delta_plus.nvals() > 0 {
+                let mut accum = SparseMatrix::new(GrB_BOOL, self.nrows(), self.ncols());
+                accum.mxm(None, &m.matrix, &n.delta_plus, semiring, null_mut());
+                if accum.nvals() > 0 {
+                    Some(accum)
+                } else {
+                    None
+                }
+            } else {
+                None
+            };
+
+            self.matrix
+                .mxm(mask.as_ref(), &m.matrix, &n.matrix, semiring, desc);
+            if let Some(accum) = accum {
+                self.matrix
+                    .element_wise_add(None, None, Some(&accum), GxB_ANY_PAIR_BOOL);
+            }
+        }
+    }
+
+    pub fn element_wise_add(
+        &mut self,
+        semiring: GrB_Semiring,
+        m: &DeltaMatrix,
+        n: &DeltaMatrix,
+    ) {
+        match (
+            m.delta_minus.nvals() > 0 || m.delta_plus.nvals() > 0,
+            n.delta_minus.nvals() > 0 || n.delta_plus.nvals() > 0,
+        ) {
+            (true, true) => {
+                self.matrix
+                    .element_wise_add(None, Some(&m.export()), Some(&n.export()), semiring);
+            }
+            (true, false) => {
+                self.matrix
+                    .element_wise_add(None, Some(&m.export()), Some(&n.matrix), semiring);
+            }
+            (false, true) => {
+                self.matrix
+                    .element_wise_add(None, Some(&m.matrix), Some(&n.export()), semiring);
+            }
+            (false, false) => {
+                self.matrix
+                    .element_wise_add(None, Some(&m.matrix), Some(&n.matrix), semiring);
+            }
+        }
+    }
+
+    pub fn export(&self) -> SparseMatrix {
+        let mut m = SparseMatrix::new(unsafe { GrB_BOOL }, self.nrows(), self.ncols());
+        if self.delta_minus.nvals() > 0 {
+            m.transpose(
+                Some(&self.delta_minus),
+                null_mut(),
+                Some(&self.matrix),
+                unsafe { GrB_DESC_RSCT0 },
+            );
+        } else {
+            m.transpose(None, null_mut(), Some(&self.matrix), unsafe {
+                GrB_DESC_RT0
+            });
+        }
+        if self.delta_plus.nvals() > 0 {
+            m.element_wise_add(None, None, Some(&self.delta_plus), unsafe {
+                GxB_ANY_PAIR_BOOL
+            });
+        }
+
+        m
+    }
+
+    pub fn pending(&self) -> bool {
+        if let Some(t) = self.transposed.as_ref() {
+            if t.pending() {
```

Comment:
I think a map() is more streamlined here.
if self.transposed.as_ref().map(|transposed_mat| transposed_mat.pending()).unwrap_or_default() {
  return true
}